widgets/heatmap/heatmapdemo/heatmapdemo.go

```diff
@@ -24,21 +24,86 @@ import (
 	"github.com/mum4k/termdash/terminal/tcell"
 	"github.com/mum4k/termdash/terminal/terminalapi"
 	"github.com/mum4k/termdash/widgets/heatmap"
+	"math/rand"
+	"time"
 )
 
+// playHeatMap continuously changes the displayed values on the heat map once every delay.
+// Exits when the context expires.
+func playHeatMap(ctx context.Context, hp *heatmap.HeatMap, delay time.Duration) {
+	const max = 100
+
+	ticker := time.NewTicker(delay)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			maxRows, maxCols := hp.ValueCapacity()
+			rows := rand.Intn(maxRows) + 1
+			cols := rand.Intn(maxCols) + 1
+
+			var values [][]float64
+			for i := 0; i < rows; i++ {
+				rv := make([]float64, cols)
+				for j := 0; j < cols; j++ {
+					rv = append(rv, float64(rand.Int31n(max+1)))
+				}
+				values = append(values, rv)
+			}
+
+			if err := hp.Values(nil, nil, values); err != nil {
+				panic(err)
+			}
+		case <-ctx.Done():
+			return
+		}
+	}
+}
+
+func getData() ([]string, []string, [][]float64) {
+	var xl = []string{
+		"one",
+		"two",
+		"three",
+		"four",
+		"five",
+		"six",
+	}
+	var yl = []string{
+		"one",
+		"two",
+		"three",
+		"four",
+		"five",
+	}
+	var values = [][]float64{
+		{1, 2, 3, 4, 5, 6},
+		{-1, -2, -3, -4, -5, -6},
+		{7, 8, 9, 10, 11, 12},
+		{12, 11, 10, 9, 8, 7},
+		{6, 5, 4, 3, 2, 1},
+	}
+	return xl, yl, values
+}
+
 func main() {
 	t, err := tcell.New()
 	if err != nil {
 		panic(err)
 	}
 	defer t.Close()
 
-	hp, err := heatmap.New()
+	hp, err := heatmap.New(heatmap.HideXLabels(), heatmap.HideYLabels())
 	if err != nil {
 		panic(err)
 	}
 
-	// TODO: set heatmap's data
+	if err := hp.Values(getData()); err != nil {
+		panic(err)
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	//go playHeatMap(ctx, hp, 1*time.Second)
```

Comment:
You might already be aware of this, the current implementation, the current implementation seems to have a bug. When we uncomment this line, we get the following failure after a couple of updates:

```go
panic: container.Draw => error: unable to draw widget *heatmap.HeatMap: unable to set cell runes in area (74,0)-(86,1), it must fit inside the available cell area is (0,0)-(78,23)

goroutine 1 [running]:
main.main()
	/Users/mum4k/go/src/github.com/mum4k/termdash/widgets/heatmap/heatmapdemo/heatmapdemo.go:127 +0x88c
exit status 2
```

Reply:
Resolved